data/json/recipes/practice/melee.json

```
@@ -149,144 +149,932 @@
+      { "proficiency": "prof_short_swords_familiar", "required": true },
+      { "proficiency": "prof_short_swords_pro", "time_multiplier": 2, "skill_penalty": 0 }
+    ],
+    "tools": [ [ "sword_xiphos", "khopesh", "machete", "survivor_machete" ], [ "water", "water_clean" ] ],
+    "//1": "Standard Blade Training Cost",
+    "components": [ [ [ "duct_tape", 10 ] ], [ [ "stick", 1 ], [ "2x4", 1 ] ], [ [ "paper", 10 ] ] ]
+  },
+  {
+    "id": "prac_bashing_beg_staff",
+    "type": "practice",
+    "activity_level": "BRISK_EXERCISE",
+    "category": "CC_PRACTICE",
+    "subcategory": "CSC_PRACTICE_MELEE",
+    "name": "quarterstaves (beginner)",
+    "description": "Practice swinging your staff, getting used to the balance and weight.",
+    "skill_used": "bashing",
+    "skills_required": [ [ "melee", 1 ] ],
+    "time": "1 h",
+    "practice_data": { "min_difficulty": 0, "max_difficulty": 1, "skill_limit": 2 },
+    "autolearn": [ [ "bashing", 2 ] ],
+    "book_learn": [ [ "mag_bashing", 1 ], [ "manual_bashing", 0 ] ],
+    "proficiencies": [ { "proficiency": "prof_quarterstaves_familiar", "time_multiplier": 1.5, "skill_penalty": 0 } ],
+    "tools": [ [ "bo", "q_staff", "i_staff", "mop", "stick_long" ] ]
+  },
+  {
+    "id": "prac_bashing_int_staff",
+    "type": "practice",
+    "activity_level": "BRISK_EXERCISE",
+    "category": "CC_PRACTICE",
+    "subcategory": "CSC_PRACTICE_MELEE",
+    "name": "quarterstaves (intermediate)",
+    "description": "Practice using your staff against a dummy, familiarizing yourself with its length.",
+    "skill_used": "bashing",
+    "skills_required": [ [ "melee", 1 ] ],
+    "time": "1 h",
+    "practice_data": { "min_difficulty": 2, "max_difficulty": 3, "skill_limit": 4 },
+    "autolearn": [ [ "bashing", 4 ] ],
+    "book_learn": [ [ "mag_bashing", 3 ], [ "manual_bashing", 2 ] ],
+    "proficiencies": [
+      { "proficiency": "prof_quarterstaves_familiar", "required": true },
+      { "proficiency": "prof_quarterstaves_pro", "time_multiplier": 2, "skill_penalty": 0 }
+    ],
+    "tools": [
+      [ "bo", "q_staff", "i_staff", "mop", "stick_long" ],
+      [ "pseudo_training_dummy_light", "pseudo_training_dummy_heavy" ]
+    ]
   },
   {
     "id": "prac_stabbing_beg_fencing",
     "type": "practice",
     "activity_level": "BRISK_EXERCISE",
     "category": "CC_PRACTICE",
     "subcategory": "CSC_PRACTICE_MELEE",
-    "name": "fencing",
+    "name": "fencing (beginner)",
     "description": "Practice the basics of the noble art of fencing.  Learn how to angle your body, aim your lunges, and parry pretend swords.  Not that the undead are using swords.  Hopefully.",
     "skill_used": "stabbing",
     "skills_required": [ [ "melee", 1 ] ],
     "time": "1 h",
     "practice_data": { "min_difficulty": 0, "max_difficulty": 1, "skill_limit": 2 },
     "autolearn": [ [ "stabbing", 2 ] ],
-    "book_learn": [ [ "mag_stabbing", 0 ] ],
+    "book_learn": [ [ "mag_stabbing", 1 ], [ "manual_fencing", 0 ] ],
```

```suggestion
    "book_learn": [ [ "mag_stabbing", 1 ] ],
```

I think think was removed in the last PR since it didn't work.

---

Whoops, my bad.